### ISSUE TITLE
Siphon attributes

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#2004]
A siphon attribute would 

For example, it could go something along the lines o:

"hull damage" 100
"hull siphon" 0.1

**Comment by Nechochwen-D** [comment]
Could you clarify a bit?

**Comment by Amazinite** [comment]
Accidentally hit post before finishing.

**Comment by Nechochwen-D** [comment]
I have read the edited version but it still is a bit confusing.

**Comment by DingusShingleton** [comment]
For whatever damage type is specified (hull, shield, ion, disruption, slowing, etc), a certain amount of damage dealt of that type will be returned to the firing ship. So if I have a ship with a hull siphon weapon, I can basically steal hull from enemy ships.

**Comment by Amazinite** [comment]
(Using your PR as an example) Say you want to create a weapon that transfers fuel from one ship to another. Right now you might do something like this:
```
"fuel damage" -100
"firing fuel" 100
```

The problem is that this would remove fuel from the firing ship even if no target was hit, and therefore no fuel was actually transferred. But with a "fuel siphon" attribute, the fuel would only be taken from the firing ship if a target is hit. You could also go in reverse and steal fuel from other ships using this attribute.

**Comment by Nechochwen-D** [comment]
I will see if this is possible.

**Comment by Elyssaen** [comment]
Hmm, I'm not sure that there's a nice link back from projectile to the parent ship that fired it. One would be needed, but beyond that it wouldn't in principle be hard.

**Comment by KiLEdEnNis** [comment]
I would get shield & energy drain, not hull, heat, fuel, disruption etc. but enabling all that will give modders more fun
and if I manage to finish my 1st ship in my project (needs to look good top-down) and I continue the project I could use shield draining

**Comment by Amazinite** [comment]
@Elyssaen agreed, although I don't think `"hit force siphon"` makes much sense as a name in that case. XD

**Comment by Nechochwen-D** [comment]
I think something along the lines of an "on hit" node would be more appropriate.

**Comment by KiLEdEnNis** [comment]
if you have "hull steal" you can make organic (actual full organic) ships and have them heal off of damage o:

**Comment by Bladewood** [comment]
outfit "Leech Missile Launcher"
		category "Secondary Weapons"
		cost 720000
		"mass" 15
		thumbnail "outfit/leech launcher"
		"outfit space" -24
		"weapon capacity" -24
		"gun ports" -1
		"leech capacity" 30
		weapon
			sprite "projectile/leech"
				"no repeat"
				"frame rate" .25
			sound "leech"
			ammo "Leech Missile"
			icon "icon/leech"
			"fire effect" "leech fire"
			"die effect" "missile death"
			"hit effect" "medium explosion"
			"inaccuracy" 12
			"velocity" 12
			"lifetime" 500
			"reload" 40
			"firing energy" 30
			"firing heat" 24
			"acceleration" 1
			"drag" .1
			"turn" 2.4
			"homing" 3
			"infrared tracking" .9
			"optical tracking" .4
			"shield damage" 360
			"hull damage" 60
			siphon
				energy 60
			"hit force" 30
			"missile strength" 42
		description "Leech Missiles exploit some obscure physical phenomenon to instantaneously sap a not-inconsiderable amount of energy from any ship it strikes, transferring it to the Leech Missile Launcher and by extension the rest of the ship. Even without their energy-siphoning ability, leech missiles are nonetheless formidable weapons, though they are relatively ineffective against hull."

	outfit "Leech Missile"
		category "Ammunition"
		cost 6000
		thumbnail "outfit/leech"
		"mass" .3
		"leech capacity" -1
		description "This missile is fired from Leech Missile Launchers. Leeches transfer energy from any ship they strike. Each launcher can hold up to 30 of them."

**Comment by warp-core** [comment]
> if you have "hull steal" you can make organic (actual full organic) ships and have them heal off of damage o:

Void Sprites gonna eat my hull to heal themselves?

**Comment by Zitchas** [comment]
> > if you have "hull steal" you can make organic (actual full organic) ships and have them heal off of damage o:
> 
> Void Sprites gonna eat my hull to heal themselves?

That would be awesome. Even better would be to give this to the Embershades. They're supposed to be predatory.
